compiler/rustc_resolve/src/lib.rs

```diff
@@ -1882,9 +1882,7 @@ impl<'a> Resolver<'a> {
 
         match self.maybe_resolve_path(&segments, Some(ns), &parent_scope) {
             PathResult::Module(ModuleOrUniformRoot::Module(module)) => Some(module.res().unwrap()),
-            PathResult::NonModule(path_res) if path_res.unresolved_segments() == 0 => {
-                Some(path_res.base_res())
-            }
+            PathResult::NonModule(path_res) if let Some(res) = path_res.full_res() => Some(res),
```

Comment:
```suggestion
            PathResult::NonModule(path_res) => path_res.full_res(),
```
The other case just returns `None`.